packages/ui/src/theme/utils.ts

```diff
@@ -48,3 +51,95 @@ interface NameTransformProps {
 export function cssNameTransform({ path = [] }: NameTransformProps): string {
   return `${kebabCase([CSS_VARIABLE_PREFIX, ...path].join(' '))}`;
 }
+
+export type pathObject = {
+  path: string[];
+};
+
+export const STATE_NAMES = [
+  'hover',
+  'focus',
+  'active',
+  'loading',
+  'disabled',
+  'visited',
+];
+
+export function findDeprecatedStateTokens(baseTheme: BaseTheme): pathObject[] {
+  const baseEntry = baseTheme?.tokens || {};
+  return traverse(['', baseEntry], []);
+}
+
+/**
+ * This function will recursively traverse the tokens object from a theme and return a path array to
+ * tokens with the deprecatedStateToken flag. This will later be used to match up against the passed in theme object
+ * and write State token values that are being passed in using the old deprecated state tokens.
+ */
+function traverse(baseEntry, existingPath: string[]): pathObject[] {
```

Comment:
nit: is this name too generic? What about `tranverseForDepracated`?

also, instead of `existingPath`, would it be more understandable to name it as `childrenPath`?